data-prepper-plugins/otel-trace-raw-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltrace/model/TraceIdTraceGroupCache.java

```diff
@@ -0,0 +1,35 @@
+package com.amazon.dataprepper.plugins.prepper.oteltrace.model;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.concurrent.TimeUnit;
+
+public class TraceIdTraceGroupCache {
```

Comment:
Do we need a separate class for this (instead of just using the Guava cache directly)?

Reply:
We could. I separate this class for (1) testability (2) we might still replace Guava cache in the future and it might incur more configuration parameters.

Reply:
Gotcha. Then for now I'd suggest just using the cache directly without a wrapper. They [already test extensively](https://github.com/google/guava/blob/master/guava-tests/test/com/google/common/cache/LocalCacheTest.java), and  currently there isn't any added functionality gained at the cost of more abstraction that I can see.

Definitely agree on the advantage of being able to swap out the backing cache later. If we want to do that then we'll just need to swap out the 7 lines of code in OTelTraceRawPrepper to get back to using a wrapper of our own.

That being said, if still you prefer to keep this wrapper then consider changing the name to make it less coupled to this specific usecase. Could use generics to let the caller decide what the cache will store instead of hardcoding the <String, String> implementation too.